### PR TITLE
[Aptos Framework] Add borrow_mut_with_default to IterableTable 

### PR DESCRIPTION
**Pull request by bcerecer**
### Description
Changes:
- Updated misleading doc for IterableTable's bottow_iter_mut
- Added borrow_mut_with_default to IterableTable 

### Test Plan
Tested locally (shouldn't generate a problem since it's reusing module's functions)

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/1792)

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/IterableTable.move [hunk 1]
```diff
@@ -107,7 +107,7 @@ module AptosFramework::IterableTable {
         (&v.val, v.prev, v.next)
     }
 
-    /// Acquire an immutable reference to the value and previous/next key which `key` maps to
+    /// Acquire an mutable reference to the value and previous/next key which `key` maps to.
```
Comment (wrwg) [inline aptos-move/framework/aptos-framework/sources/IterableTable.move:119]: *a* mutable

**Review by wrwg — APPROVED** [review]
Seeing design of this iterable for the first time, looks very nice and avoids the issues of HashTable or OrderedTable regards iteration order.

File: aptos-move/framework/aptos-framework/sources/IterableTable.move [hunk 2]
```diff
@@ -73,15 +73,6 @@ module AptosFramework::IterableTable {
         &mut Table::borrow_mut(&mut table.inner, key).val
     }
 
-    /// Acquire a mutable reference to the value which `key` maps to.
```
Comment (sitalkedia) [inline aptos-move/framework/aptos-framework/sources/IterableTable.move]: Seems like you accidentally removed this function in your latest commit?

**Review by sitalkedia — COMMENTED** [review]
Seems like you accidentally removed the function `borrow_mut_with_default` in your latest commit?

File: aptos-move/framework/aptos-framework/sources/IterableTable.move (hunk 2 above)
Comment (bcerecer) [reply aptos-move/framework/aptos-framework/sources/IterableTable.move]: You're correct, I brought it back

**Comment by banool** [comment]
I had a note lying around saying that we should have this, and it seems you've done it. Thanks @bcerecer!

**Pull request merged**